axum-macros/src/debug_handler.rs

```diff
@@ -39,29 +39,55 @@ pub(crate) fn expand(attr: Attrs, item_fn: ItemFn) -> TokenStream {
     }
 }
 
+mod kw {
+    syn::custom_keyword!(body);
+    syn::custom_keyword!(state);
+}
+
 pub(crate) struct Attrs {
     body_ty: Type,
+    state_ty: Type,
 }
 
 impl Parse for Attrs {
     fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
         let mut body_ty = None;
+        let mut state_ty = None;
 
         while !input.is_empty() {
-            let ident = input.parse::<syn::Ident>()?;
-            if ident == "body" {
+            let lh = input.lookahead1();
```

Comment:
One day we should refactor the argument parsing to be done the same way for each macro, but today is not that day 😅